core/lib/via_btc_client/src/indexer/mod.rs

```diff
@@ -217,27 +285,331 @@ impl BitcoinInscriptionIndexer {
             }
-            _ => {} // ignore other messages
+            _ => {
+                debug!("Ignoring non-bootstrap message during bootstrap process");
+            }
         }
     }
 
-    fn get_sender_address(common_fields: &CommonFields) -> Option<Address> {
+    #[instrument(skip(self, message), target = "bitcoin_indexer")]
+    fn is_valid_l1_to_l2_transfer(&self, message: &L1ToL2Message) -> bool {
+        let is_valid = message
+            .tx_outputs
+            .iter()
+            .any(|output| output.script_pubkey == self.bridge_address.script_pubkey());
+        debug!("L1ToL2Message transfer validity: {}", is_valid);
+        is_valid
+    }
+
+    #[instrument(skip(common_fields), target = "bitcoin_indexer")]
+    fn get_sender_address(common_fields: &CommonFields, network: Network) -> Option<Address> {
         secp256k1::XOnlyPublicKey::from_slice(common_fields.encoded_public_key.as_bytes())
             .ok()
             .map(|public_key| {
                 Address::p2tr(
                     &bitcoin::secp256k1::Secp256k1::new(),
                     public_key,
                     None,
-                    KnownHrp::from(Network::Testnet), // TODO: make it configurable
+                    KnownHrp::from(network),
                 )
             })
     }
+}
 
-    fn is_valid_l1_to_l2_transfer(&self, message: &L1ToL2Message) -> bool {
-        message
-            .tx_outputs
-            .iter()
-            .any(|output| output.script_pubkey == self.bridge_address.script_pubkey())
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use bitcoin::{
+        block::Header, hashes::Hash, Amount, Block, OutPoint, ScriptBuf, Transaction, TxMerkleNode,
+        TxOut,
+    };
+    use mockall::{mock, predicate::*};
+    use secp256k1::Secp256k1;
+
+    use super::*;
+    use crate::types::BitcoinClientResult;
+
+    mock! {
+        BitcoinOps {}
+        #[async_trait]
+        impl BitcoinOps for BitcoinOps {
+            async fn get_transaction(&self, txid: &Txid) -> BitcoinClientResult<Transaction>;
+            async fn fetch_block(&self, block_height: u128) -> BitcoinClientResult<Block>;
+            async fn fetch_block_by_hash(&self, block_hash: &BlockHash) -> BitcoinClientResult<Block>;
+            async fn get_balance(&self, address: &Address) -> BitcoinClientResult<u128>;
+            async fn broadcast_signed_transaction(&self, signed_transaction: &str) -> BitcoinClientResult<Txid>;
+            async fn fetch_utxos(&self, address: &Address) -> BitcoinClientResult<Vec<(OutPoint, TxOut)>>;
+            async fn check_tx_confirmation(&self, txid: &Txid, conf_num: u32) -> BitcoinClientResult<bool>;
+            async fn fetch_block_height(&self) -> BitcoinClientResult<u128>;
+            async fn get_fee_rate(&self, conf_target: u16) -> BitcoinClientResult<u64>;
+            fn get_network(&self) -> Network;
+        }
+    }
+
+    fn create_mock_indexer() -> BitcoinInscriptionIndexer {
```

Comment:
Instead of creating the `mock_client` inside of this function you can pass it as an argument and then you can reuse it for all tests.